src/config/mongoose_config_spec.erl

```diff
@@ -1154,19 +1197,23 @@ check_auth_method(Method, Opts) ->
         false -> error(#{what => missing_section_for_auth_method, auth_method => Method})
     end.
 
-process_pool([Tag, Type|_], KVs) ->
-    {[ScopeOpts, HostOpts, ConnOpts], Opts} = proplists:split(KVs, [scope, host, connection]),
-    Scope = pool_scope(ScopeOpts, HostOpts),
-    Connection = pool_connection(ConnOpts),
-    {b2a(Type), Scope, b2a(Tag), Opts, Connection}.
-
-pool_scope([{scope, single_host}], [{host, Host}]) -> Host;
-pool_scope([{scope, host}], []) -> host;
-pool_scope([{scope, global}], []) -> global;
-pool_scope([], []) -> global.
-
-pool_connection([{connection, Opts}]) -> Opts;
-pool_connection([]) -> [].
+process_pool([Tag, Type|_], AllOpts) ->
+    Scope = pool_scope(maps:get(scope, AllOpts, none), maps:get(host, AllOpts, none)),
+    Connection = maps:get(connection, AllOpts, #{}),
+    Opts = maps:without([scope, host, connection], AllOpts),
+    #{type => b2a(Type),
+      scope => Scope,
+      tag => b2a(Tag),
+      opts => Opts,
+      conn_opts => Connection}.
+
+pool_scope(single_host, none) ->
+    error(#{what => pool_single_host_not_specified,
+            text => <<"\"host\" option is required if \"single_host\" is used.">>});
+pool_scope(single_host, Host) -> Host;
+pool_scope(host, none) -> host;
+pool_scope(global, none) -> global;
+pool_scope(none, none) -> global.
```

Comment:
I think we can have a default specified for `scope` instead.

Reply:
You mean `host`, right?

Reply:
I meant `scope = host`

Reply:
Ahhh, the _default_ default, I see. OK